hivelvet-backend/tests/src/Actions/Roles/EditTest.php

```diff
@@ -20,98 +20,81 @@
  * with Hivelvet; if not, see <http://www.gnu.org/licenses/>.
  */
 
-namespace Models;
+namespace Actions\Roles;
 
 use Fake\RoleFaker;
+use ReflectionException;
 use Test\Scenario;
 
 /**
- * Class UserTest.
- *
  * @internal
  * @coversNothing
  */
-final class RoleTest extends Scenario
+final class EditTest extends Scenario
 {
-    final protected const ADD_ROLE_ROUTE    = 'POST /roles/add';
-    final protected const EDIT_ROLE_ROUTE   = 'PUT /roles/edit/';
-    final protected const DELETE_ROLE_ROUTE = 'DELETE /roles/delete/';
-
-    protected $group = 'Role Model';
+    final protected const EDIT_ROLE_ROUTE = 'PUT /roles/edit/';
+    protected $group                      = 'Action Role Edit';
 
     /**
-     * @param \Base $f3
+     * @param $f3
+     *
+     * @throws ReflectionException
      *
      * @return array
      */
-    public function testRoleCreation($f3)
+    public function testEmptyName($f3)
     {
         $test = $this->newTest();
 
         $role = RoleFaker::create();
-        $test->expect($role->valid(), 'Role mocked & saved to the database');
-
         $data = ['data' => ['name' => '']];
-        $f3->mock(self::ADD_ROLE_ROUTE, null, null, $this->postJsonData($data));
-        $test->expect($this->compareTemplateToResponse('role/empty_error.json'), 'Add role with empty name show an error');
-
-        $data = ['data' => ['name' => $role->name]];
-        $f3->mock(self::ADD_ROLE_ROUTE, null, null, $this->postJsonData($data));
-        $test->expect($this->compareTemplateToResponse('role/exist_error.json'), 'Add role with existing name show an error');
+        $f3->mock(self::EDIT_ROLE_ROUTE . $role->id, null, null, $this->postJsonData($data));
+        $test->expect($this->compareTemplateToResponse('role/empty_error.json'), 'Update role with empty name show an error');
 
         return $test->results();
     }
 
     /**
-     * @param \Base $f3
+     * @param $f3
+     *
+     * @throws ReflectionException
      *
      * @return array
      */
-    public function testRoleUpdate($f3)
+    public function testExistingName($f3)
     {
         $test = $this->newTest();
 
-        $role = RoleFaker::create();
-        $test->expect($role->valid(), 'Role mocked & saved to the database');
-
-        $role2 = RoleFaker::create();
-        $test->expect($role2->valid(), 'Role 2 mocked & saved to the database');
-
-        $data = ['data' => ['name' => '']];
-        $f3->mock(self::EDIT_ROLE_ROUTE . $role->id, null, null, $this->postJsonData($data));
-        $test->expect($this->compareTemplateToResponse('role/empty_error.json'), 'Update role with empty name show an error');
-
-        $data = ['data' => ['name' => $role2->name]];
-        $f3->mock(self::EDIT_ROLE_ROUTE . $role->id, null, null, $this->postJsonData($data));
+        $faker_1 = RoleFaker::create();
```

Comment:
I think there is a confusion here between a `faker` and a `role` entity. Since a faker creates an entity, the naming of the variable should be `$role.`